app/controllers/v1/submissions_controller.rb

```diff
@@ -12,6 +12,11 @@ def show
   def create
     task = Task.find(params.dig(:submission, :task_id))
 
+    if !correcting_submission? && (submission = task.active_submission)&.completed?
```

Comment:
I found this line difficult to parse. The assignment in the conditional doesn't help. Perhaps it could be made easier to follow without it, e.g. `if !correcting_submission? && task.active_submission&.completed?` and then reference `task.active_submission` on the `render` line? Or maybe even extract it to its own method. `trying_to_correct_an_incomplete_task?` or something?